packages/block-editor/src/components/block-controls/index.js

```diff
@@ -44,7 +45,36 @@ function BlockControlsFill( { controls, children } ) {
 	);
 }
 
-const BlockControls = ifBlockEditSelected( BlockControlsFill );
+function BlockControls( { __experimentalAllowAnyMultiple, children } ) {
+	const { isSelected, clientId, name } = useBlockEditContext();
+	const isFirstAndSameTypeMultiSelected = useSelect(
+		( select ) => {
+			const {
+				getBlockName,
+				isFirstMultiSelectedBlock,
+				getMultiSelectedBlockClientIds,
+			} = select( 'core/block-editor' );
+
+			if ( ! isFirstMultiSelectedBlock( clientId ) ) {
+				return false;
```

Comment:
Why `return false;` and not `return null;` or even `return;`?

Reply:
Because `isFirstAndSameTypeMultiSelected` is meant to be boolean